diff --git a/gson/src/main/java/com/google/gson/stream/JsonWriter.java b/gson/src/main/java/com/google/gson/stream/JsonWriter.java
index 6a935dd..e00da06 100644
--- a/gson/src/main/java/com/google/gson/stream/JsonWriter.java
+++ b/gson/src/main/java/com/google/gson/stream/JsonWriter.java
@@ -399,7 +399,7 @@ public class JsonWriter implements Closeable, Flushable {
   private void writeDeferredName() throws IOException {
     if (deferredName != null) {
       beforeName();
-      string(deferredName);
+      out.write(deferredName);
       deferredName = null;
     }
   }
